pytorch_lightning/trainer/trainer.py

```diff
@@ -1560,7 +1559,7 @@ def _call_callback_hooks(
             for callback in self.callbacks:
                 fn = getattr(callback, hook_name)
                 if callable(fn):
-                    with self.profiler.profile(hook_name):
+                    with self.profiler.profile(f"{callback.__class__.__name__}.{callback.state_key}.{hook_name}"):
```

Comment:
same here -`Callback.{state_key}.{hook_name}` ?

also `callback.__class__.__name__` could be redundant w/ state_key. 

State key should include the qualname already. 
https://github.com/PyTorchLightning/pytorch-lightning/blob/5576fbc5f9a7d0bc71ad26b8b54775110e675808/pytorch_lightning/callbacks/base.py#L36-L58


Here's an example from the early stopping callback: https://github.com/PyTorchLightning/pytorch-lightning/blob/5576fbc5f9a7d0bc71ad26b8b54775110e675808/pytorch_lightning/callbacks/early_stopping.py#L124-L126